tests/integration_tests/data_steward/retraction/retract_deactivated_pids_test.py

```diff
@@ -0,0 +1,283 @@
+            result = response.result()
+            if query_dict['destination_table_id'] in expected_dropped_row_count:
+                expected_dropped_row_count[
+                    query_dict['destination_table_id']] += result.total_rows
+            else:
+                expected_dropped_row_count[
+                    query_dict['destination_table_id']] = result.total_rows
+
+        # Separate check to find number of actual deleted rows
+        q = TABLE_ROWS_QUERY.format(dataset_id=self.bq_dataset_id)
+        q_result = self.client.query(q)
+        row_count_before_retraction = {}
+        for row in q_result:
+            row_count_before_retraction[row['table_id']] = row['row_count']
+
+        # Use query results to count number of expected dropped row deletions
+        expected_kept_row_count = {}
+        for query_dict in kept_row_count_queries:
+            response = self.client.query(query_dict['query'])
+            result = response.result()
+            if query_dict['destination_table_id'] in expected_kept_row_count:
+                expected_kept_row_count[query_dict['destination_table_id']] -= (
+                    (row_count_before_retraction[
+                        query_dict['destination_table_id']] -
+                     result.total_rows))
+            else:
+                expected_kept_row_count[query_dict['destination_table_id']] = (
+                    row_count_before_retraction[
+                        query_dict['destination_table_id']] -
+                    (row_count_before_retraction[
+                        query_dict['destination_table_id']] -
+                     result.total_rows))
+
+        # Perform retraction
+        query_list = retract_deactivated_pids.create_queries(
+            self.project_id, self.ticket_number, self.project_id,
+            self.bq_dataset_id, self.pid_table_id)
+        retract_deactivated_pids.run_queries(query_list, self.client)
+
+        # Find actual deleted rows
+        q_result = self.client.query(q)
+        results = q_result.result()
+        row_count_after_retraction = {}
+        for row in results:
+            row_count_after_retraction[row['table_id']] = row['row_count']
+
+        for table in expected_dropped_row_count:
+            self.assertEqual(
+                expected_dropped_row_count[table],
+                row_count_before_retraction[table] -
+                row_count_after_retraction[table])
+
+        for table in expected_kept_row_count:
+            self.assertEqual(expected_kept_row_count[table],
+                             row_count_after_retraction[table])
+
+    def tearDown(self):
+        test_util.delete_all_tables(self.bq_dataset_id)
+        test_util.delete_all_tables(self.bq_sandbox_dataset_id)
+        tables = list(self.client.list_tables(self.bq_sandbox_dataset_id))
+        if not tables:
+            self.client.delete_dataset(self.bq_sandbox_dataset_id)
```

Comment:
This check is not strictly necessary.  The test_util is going to delete every table in the sandbox dataset when you call it.  You should be fine to just remove the sandbox dataset after that.  If a test left data in there, it shouldn't have.

Reply:
Ok great, I will remove this check, thanks!